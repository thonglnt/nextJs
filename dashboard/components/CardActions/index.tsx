import { Typography, TypographyProps, useTheme } from '@mui/material'
import Link from 'next/link'
import { memo } from 'react'

export interface ICardActions extends TypographyProps {
  title: string
  href: string
}
const CardActions = ({ title, href, ...props }: ICardActions) => {
  const theme = useTheme()

  return (
    <Typography
      color={theme.palette.secondary.main}
      fontSize="13px"
      lineHeight="22px"
      {...props}
    >
      <Link href={href} prefetch={false}>
        {title}
      </Link>
    </Typography>
  )
}

export default memo(CardActions)
