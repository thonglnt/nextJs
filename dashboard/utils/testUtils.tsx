import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
