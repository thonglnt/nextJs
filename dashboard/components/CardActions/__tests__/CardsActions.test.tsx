// Utils
import { render, screen } from '@utils/testUtils'

// Components
import CardActions, { ICardActions } from '..'

describe('Card Actions component', () => {
  const mockProps: ICardActions = {
    title: 'Learn more',
    href: '/learn-more',
  }

  it('should render Card Actions component', () => {
    const { container } = render(<CardActions {...mockProps} />)

    expect(container).toMatchSnapshot()
  })

  it('should render match content Card Actions component', () => {
    render(<CardActions {...mockProps} />)

    const contentCardActions = screen.getByText(/Learn more/i)
    expect(contentCardActions).toHaveTextContent('Learn more')
  })
})
