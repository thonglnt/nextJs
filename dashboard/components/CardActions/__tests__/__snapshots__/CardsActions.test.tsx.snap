// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Actions component should render Card Actions component 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1hazt9m-MuiTypography-root"
  >
    <a
      href="/learn-more"
    >
      Learn more
    </a>
  </p>
</div>
`;
